### PR TITLE
Bump OTel to 1.4.0

### LINKED ISSUE #1027: Remove DynamicMethod dependent sources from IncludeSharedInstrumentationSource
Separate PR:
Exclude from `IncludeSharedInstrumentationSource` files which depends on DynamicMethod.
Crate separate Property to include onlye DynamibMethod based clases do this, revert changes in this file.

_Originally posted by @Kielek in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1020#discussion_r1113960154_

### PR DESCRIPTION
**Pull request by Kielek**
## Changes

* Bump OTel 1.4.0 together with non-Core packages.
* `Exporter.Instana` was pinned to 1.3.2. For now it is referencing net4.6.1
* Adjust tests.
* Tests executed only against netcoreapp3.1 are executed also on net7.0 and net6.0 (projects referencing OTel 1.4.0).

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* ~~[ ] Design discussion issue #~~
* ~~[ ] Changes in public API reviewed~~

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Instana/OpenTelemetry.Exporter.Instana.csproj
Comment (pellared) [inline src/OpenTelemetry.Exporter.Instana/OpenTelemetry.Exporter.Instana.csproj:28]: I suggest creating an issue for it.

**Review by pellared — APPROVED** [review]

File: src/OpenTelemetry.Exporter.Instana/OpenTelemetry.Exporter.Instana.csproj
Comment (Kielek) [reply src/OpenTelemetry.Exporter.Instana/OpenTelemetry.Exporter.Instana.csproj:28]: It should be part of #1031

**Review by pellared — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
_Drop netcoreapp3.1 tests for projects based on OpenTelemetryCoreLatestVersion (to avoid compilation warnings)_

^ We need to either revert this part, OR work-around the compiler warning, OR Get explicit approval from the component owners before merge.

File: src/OpenTelemetry.Contrib.Instrumentation.AWS/README.md
Comment (utpilla) [inline src/OpenTelemetry.Contrib.Instrumentation.AWS/README.md:20]: I'm not sure if we should simply update the README with the new methods.

It could be the case that a lot of the users of AWS instrumentation still use older version of OTel and the extensions hosting package (consider this project still references OTel SDK 1.0.1)

**Comment by Kielek** [comment]
> _Drop netcoreapp3.1 tests for projects based on OpenTelemetryCoreLatestVersion (to avoid compilation warnings)_
> 
> ^ We need to either revert this part, OR work-around the compiler warning, OR Get explicit approval from the component owners before merge.

I will check the workaround. As I remember there is some switch for it. Give me around one hour

File: src/OpenTelemetry.Contrib.Instrumentation.AWS/README.md
Comment (Kielek) [reply src/OpenTelemetry.Contrib.Instrumentation.AWS/README.md:20]: I will revert this file. Other redmes looks good

File: src/OpenTelemetry.Instrumentation.StackExchangeRedis/CHANGELOG.md
Comment (vishweshbankwar) [inline src/OpenTelemetry.Instrumentation.StackExchangeRedis/CHANGELOG.md:9]:
```suggestion
  ([#1038](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1038))
```

Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.StackExchangeRedis/CHANGELOG.md]: 94c52e5f615f5cb855a477ac840e584331780c3b

File: src/OpenTelemetry.Contrib.Instrumentation.AWS/README.md
Comment (Kielek) [reply src/OpenTelemetry.Contrib.Instrumentation.AWS/README.md]: 834f46717bd61194b29f14cca4ae497c2672d18c

File: src/OpenTelemetry.Instrumentation.StackExchangeRedis/CHANGELOG.md
Comment (vishweshbankwar) [inline src/OpenTelemetry.Instrumentation.StackExchangeRedis/CHANGELOG.md:7]: @Kielek Could you please clarify why we need to add this?

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.StackExchangeRedis/CHANGELOG.md:7]: I got my answer.

Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.StackExchangeRedis/CHANGELOG.md:7]: I think it is good practice to document any new dependency.
The same was done for OTLP package https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md#140-rc4

**Review by cijothomas — APPROVED** [review]

**Review by utpilla — APPROVED** [review]

**Review by mic-max — COMMENTED** [review]
LGTM wrt EventCounters 👍

**Pull request merged**
